### PR TITLE
Allow creation of metric events via backend API

### PR DESCRIPTION
**Pull request by palourde**
## What is this change?

It allow users to use the backend API to submit events containing metrics, without a check.

Additionally, I removed the `Update` method for the Events controller, which was no longer used.

## Why is this change necessary?

Closes https://github.com/sensu/sensu-go/issues/1688

## Does your change need a Changelog entry?

Added!

## Do you need clarification on anything?

Nope!

## Were there any complications while making this change?

Nope!

## Have you reviewed and updated the documentation for this change? Is new documentation required?

Nope!

### CODE REVIEW

File: testing/e2e/main_test.go [hunk 1]
```diff
@@ -19,7 +19,6 @@ var (
 	agentPortCounter                     int64 = 20000
 	agentPath, backendPath, sensuctlPath string
 	binDir                               = filepath.Join("..", "..", "bin")
-	toolsDir                             = filepath.Join(binDir, "tools")
```
Comment (echlebek) [inline testing/e2e/main_test.go]: Curious why this is in the PR, seems unrelated?

Comment (palourde) [reply testing/e2e/main_test.go]: Oh my bad, related to https://github.com/sensu/sensu-go/pull/1713, forgot to merge master back.
